contributors/design-proposals/arbitrator.md

```diff
@@ -0,0 +1,265 @@
+  namespace’s request (pending pods in PoC)
+1. Evict Pods of overused namespace
+1. Provide predicates that “namespace will not exceed Allocation.deserved” as scheduler's HTTPExtender
+
+Meanwhile, `k8s-scheduler` dispatch tasks to host based on its policy, e.g. PodAffinity:
+`k8s-arbitrator` own resource allocation, `k8s-scheduler` own resource assignment.
+
+The arbitrator takes DRF as default policy. It will list pods/nodes from `k8s-apiserver`, and calculate 
+the deserved resource of each namespace based on DRF algorithm; and then, update the deserved to `Allocation` accordingly.
+The default interval is 1s which is configurable. The arbitrator will NOT assign hostname to the deserved resources;
+it dependent on scheduler to dispatch tasks on the suitable hosts.
+
+The arbitrator also meets the following requirements:
+
+* Total deserved resource of namespace can not exceed resources in cluster
+* The deserved resources can not exceed hard resource in `Consumer`
+* The deserved resources can not less than reserved resources if enough resource in cluster
+
+### Preemption:
+When workload/quota changed, the deserved resources of each namespace maybe also changed. The “higher” priority pod will
+trigger eviction, the following figure demonstrate the case of eviction because of deserved resources.
+
+```text
+T1:                 T2:                               T3:
+ --------------     -------------- --------------     -------------- --------------
+ | Consumer-1 |     | Consumer-1 | | Consumer-2 |     | Consumer-1 | | Consumer-2 |
+ |   cpu:2    | ==> |   cpu:1    | |   cpu:0    | ==> |   cpu:1    | |   cpu:1    | 
+ |   mem:2    |     |   mem:1    | |   mem:0    |     |   mem:1    | |   mem:1    |
+ --------------     -------------- --------------     -------------- --------------
+```
+
+* T1: there's only one namespace in the cluster: `Consumer-1`; all resource(cpu:2,mem:2) are allocated to it
+* T2: a new namespace `Consuemr-2` is created; the arbitrator re-calculate resource allocation for each namespace,
+  shrink the overused namespace
+* T3: the controller, who manages overused namespace, MUST select a Pod to evict, or arbitrator will evict randomly.
+  After eviction, assign resources to underused namespace
+
+
+The arbitrator uses “/evict” subresource of pod to reclaim resources. But when arbitrator choose the pods to evict,
+here’re at least two requirements:
+* After eviction, the of pods can not less than `PodDisruptionBudget`
+* After eviction, the resources of namespace can not less than reserved
+
+The namespace maybe underused after eviction; the arbitrator will try to evict pods from most overused namespace.
+For the resources fragment issue, it’s out scope of this doc; will design detail in preemption implementation doc.
+
+## Feature Interaction:
+
+### Scheduler
+Use scheduler’s HTTPExtender to access arbitrator for predicates based on `Allocation.Deserved`.
```

Comment:
IMHO in this case it would really be nice if the http extender was configurable at runtime - and not as it's currently required at deploy time.